Framework/DataHandling/test/LoadSampleShapeTest.h

```diff
@@ -0,0 +1,174 @@
+#ifndef LOAD_SHAPETEST_H_
+#define LOAD_SHAPETEST_H_
+
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/Sample.h"
+#include "MantidDataHandling/LoadSampleShape.h"
+#include "MantidDataHandling/LoadInstrument.h"
+#include "MantidGeometry/Objects/MeshObject.h"
+#include "MantidKernel/OptionalBool.h"
+#include "MantidTestHelpers/ComponentCreationHelper.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+#include <cxxtest/TestSuite.h>
+
+using namespace Mantid;
+using namespace Mantid::API;
+using namespace Mantid::DataHandling;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+
+class LoadSampleShapeTest : public CxxTest::TestSuite {
```

Comment:
Where are the negative tests?
Missing file?
Incorrect file format, you have at least two throw lines inside the algorithm, these should be tested.

Reply:
Missing file is dealt by the file property manager when one sets the filename property, so is not necessary, but I will add tests for malformed files.